src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/controllers/CaseActionsForCaseWorkerController.java

```diff
@@ -1166,4 +1208,18 @@ private void generateEthosCaseReference(CaseData caseData, CCDRequest ccdRequest
         }
     }
 
+    private List<RepresentedTypeRItem> updateWithRespondentIds(CaseData caseData) {
+        return caseData.getRepCollection().stream()
+            .peek(respondentRep -> {
+                final List<RespondentSumTypeItem> respondentCollection = caseData.getRespondentCollection();
+                Optional<RespondentSumTypeItem> matchedRespondent = respondentCollection.stream()
+                    .filter(resp ->
+                        resp.getValue().getRespondentName()
+                            .equals(respondentRep.getValue().getRespRepName())).findFirst();
+
+                matchedRespondent.ifPresent(respondent ->
+                    respondentRep.getValue().setRespondentId(respondent.getId()));
+
+            }).collect(toList());
+    }
```

Comment:
Nested streams are really hard to read, can this be refactored to remove some complexity?

Reply:
Refactored